nvse/nvse/Hooks_Other.cpp

```diff
@@ -179,6 +180,84 @@ namespace OtherHooks
 		LambdaManager::MarkScriptAsDeleted(script);
 	}
 
+	namespace IMod {
+		void __fastcall ApplyIMOD(void** a1, void* unused, void* a2) {
+			auto *ebp = GetParentBasePtr(_AddressOfReturnAddress(), false);
+			auto ref = *reinterpret_cast<TESForm**>(ebp + 0x8);
+
+			EventManager::DispatchEvent("onapplyimod", nullptr, ref);
+
+			ThisStdCall(0x633C90, a1, a2);
+		}
+
+		void __fastcall RemoveIMOD(void** a1, void* unused, void* a2) {
+			auto* ebp = GetParentBasePtr(_AddressOfReturnAddress(), false);
+			auto ref = *reinterpret_cast<TESForm**>(ebp + 0x8);
+
+			EventManager::DispatchEvent("onremoveimod", nullptr, ref);
+
+ 			ThisStdCall(0x633C90, a1, a2);
+		}
+
+		void WriteHooks() {
+			WriteRelCall(0x529A37, reinterpret_cast<UInt32>(ApplyIMOD));
+			WriteRelCall(0x529D37, reinterpret_cast<UInt32>(RemoveIMOD));
+		}
+	}
+
+	namespace Locks {
+		void __fastcall OnLockBroken(TESObjectREFR* doorRef) {
+			EventManager::DispatchEvent("onlockbroken", nullptr, doorRef);
+
+			ThisStdCall(0x569160, doorRef);
+		}
+
+		void __fastcall OnLockPickSuccess(TESObjectREFR* doorRef) {
+			EventManager::DispatchEvent("onlockpicksuccess", nullptr, doorRef);
+
+			ThisStdCall(0x5692A0, doorRef);
+		}
+
+		void __fastcall OnLockPickBroken() {
+			auto* ebp = GetParentBasePtr(_AddressOfReturnAddress(), false);
```

Comment:
It would probably be cleaner to get the LockPickMenu from the `g_lockPickMenu` global at `0x11DA204`.

Reply:
Done, thanks